cuda_core/cuda/core/experimental/_launch_config.py

```diff
@@ -80,6 +90,13 @@ def __post_init__(self):
                     f"thread block clusters are not supported on devices with compute capability < 9.0 (got {cc})"
                 )
             self.cluster = cast_to_3_tuple("LaunchConfig.cluster", self.cluster)
+            # When cluster is set, grid should represent the number of clusters, not blocks.
+            # Convert grid dimensions from cluster units to block units by multiplying by cluster dimensions.
+            self.grid = (
+                self.grid[0] * self.cluster[0],
+                self.grid[1] * self.cluster[1], 
+                self.grid[2] * self.cluster[2]
+            )
```

Comment:
Thinking about this more, I realize placing this modification here is wrong. `LaunchConfig` is a Python dataclass. The implicit contract is: What users specify should be transparently visible and a roundtrip retrieval should not change anything. So if a user does `c = LaunchConfig(grid=..., ...)` and then `c.grid`, the output should be the same as the input regardless if cluster is set.

We should move this conversion to `_to_native_launch_config()` below:
- if `drv_cfg.cluster` is supplied, we modify grid and populate `drv_cfg.gridDim{X,Y,Z}` with the modified values
- otherwise, we populate `drv_cfg.gridDim{X,Y,Z}` with `drv_cfg.grid`, as before